### PR TITLE
[DOCS-5255] Single Source Ruby Compatibility Requirements Page

### PR DESCRIPTION
**Pull request by brett0000FF**
### What does this PR do? What is the motivation?

https://datadoghq.atlassian.net/browse/DOCS-5255
Single source new Compatibility.md page I created in [dd-trace-rb](https://github.com/DataDog/dd-trace-rb/pull/3133).

### Merge instructions

Need to juggle merging the other PR in dd-trace-rb. dd-trace-rb PR needs to merge before this one.

Before merging:

- Re-enable caching.
- Change branch back to master.
- Point dependency at master branch version of Compatibility.md.

- [ ] Please merge after reviewing

### Additional notes

- Compatibility Requirements page: [preview](https://docs-staging.datadoghq.com/brett0000FF/DOCS-5255/tracing/trace_collection/compatibility/ruby/)
- Tracing Ruby Application page: [preview](https://docs-staging.datadoghq.com/brett0000FF/DOCS-5255/tracing/trace_collection/dd_libraries/ruby/)

### CODE REVIEW

File: content/en/tracing/trace_collection/compatibility/ruby.md [hunk 1]
```diff
@@ -1,56 +1,56 @@
 ---
```
Comment (marcotc) [inline content/en/tracing/trace_collection/compatibility/ruby.md:1]: Do we still need this file, now that it is single sourced?

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 - config:
-    cache_enabled: true
+    cache_enabled: false
```
Comment (marcotc) [inline local/bin/py/build/configurations/pull_config_preview.yaml:3]: This is a global file change, just want to confirm that this is intentional.

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 3]
```diff
@@ -642,3 +642,27 @@
...
+              front_matters:
+                title: Ruby Compatibility Requirements
+                kind: documentation
+                aliases:
+                  - /tracing/compatibility_requirements/ruby
+                  - /tracing/setup_overview/compatibility_requirements/ruby
+                code_lang: ruby
+                type: multi-code-lang
+                code_lang_weight: 20
+                dependencies: ["https://github.com/DataDog/dd-trace-rb/blob/aeaf76dad192d3676d5b0db270e07708bc82782e/docs/Compatibility.md"]
```
Comment (brett0000FF) [inline local/bin/py/build/configurations/pull_config_preview.yaml:664]:
```suggestion
                dependencies: ["https://github.com/DataDog/dd-trace-rb/blob/release/docs/Compatibility.md"]
```

File: local/bin/py/build/configurations/pull_config_preview.yaml (hunk 2 above)
Comment (brett0000FF) [inline local/bin/py/build/configurations/pull_config_preview.yaml:3]:
```suggestion
    cache_enabled: true
```

File: local/bin/py/build/configurations/pull_config_preview.yaml [hunk 4]
```diff
@@ -642,3 +642,27 @@
               base_path: ""
               dest_path: "/layouts/shortcodes/"
               file_name: "semantic-color.md"
+
+      - repo_name: dd-trace-rb
+        contents:
+          - action: pull-and-push-file
+            branch: brett0000FF/DOCS-5255
```
Comment (brett0000FF) [inline local/bin/py/build/configurations/pull_config_preview.yaml:649]:
```suggestion
            branch: release
```

File: local/bin/py/build/configurations/pull_config.yaml [hunk 5]
```diff
@@ -641,3 +641,27 @@
               base_path: ""
               dest_path: "/layouts/shortcodes/"
               file_name: "semantic-color.md"
+
+        - repo_name: dd-trace-rb
+          contents:
+            - action: pull-and-push-file
+              branch: master
```
Comment (brett0000FF) [inline local/bin/py/build/configurations/pull_config.yaml:648]:
```suggestion
              branch: release
```

File: local/bin/py/build/configurations/pull_config_preview.yaml (hunk 2 above)
Comment (brett0000FF) [reply local/bin/py/build/configurations/pull_config_preview.yaml:3]: Thanks! Yeah, we temporarily change this for the preview to work, but I'll change it back before merging.

**Review by brett0000FF — COMMENTED** [review]
Apply these changes before merging.

File: content/en/tracing/trace_collection/compatibility/ruby.md (hunk 1 above)
Comment (brett0000FF) [reply content/en/tracing/trace_collection/compatibility/ruby.md]: Nope! Removed it. Thanks!

File: local/bin/py/build/configurations/pull_config_preview.yaml (hunk 2 above)
Comment (brett0000FF) [reply local/bin/py/build/configurations/pull_config_preview.yaml]: Yep, this was just a temporary change for the preview to work correctly. I reverted it back now. 👍

**Review by marcotc — APPROVED** [review]

**Comment by cswatt** [comment]
The new page seems to contain everything on the [old page](https://docs.datadoghq.com/tracing/trace_collection/compatibility/ruby/), with the exception of the paragraph and code snippet under [Integration Instrumentation](https://docs.datadoghq.com/tracing/trace_collection/compatibility/ruby/#integration-instrumentation). Is this intentional?

Otherwise everything looks good.

**Comment by brett0000FF** [comment]
> The new page seems to contain everything on the [old page](https://docs.datadoghq.com/tracing/trace_collection/compatibility/ruby/), with the exception of the paragraph and code snippet under [Integration Instrumentation](https://docs.datadoghq.com/tracing/trace_collection/compatibility/ruby/#integration-instrumentation). Is this intentional?
> 
> Otherwise everything looks good.

@ceciliawatt - Thanks for taking a look! Yeah, the pages should be very similar. They were meant to be kept in sync manually before, so there are some minor discrepancies, and now that it's single sourced, it'll stay in sync for us. For example, hanami was missing from the current page.

**Review by alai97 — APPROVED** [review]

**Pull request merged**